### ISSUE TITLE
airmon-ng: lines 134 & 137, `[ "$(find ... | wc -l)" != '0' ]` will always return true

### ISSUE THREAD

**Issue by wileyhy** [aircrack-ng/aircrack-ng#2556]
# Defect Report

## Issue type

- [y] Defect - Incorrect value displayed/received/stored

## System information

[liveuser@localhost-live]$ `cat /etc/os-release`

> NAME="Fedora Linux"
> VERSION="38 (Xfce)"
> ID=fedora
> VERSION_ID=38
> VERSION_CODENAME=""
> PLATFORM_ID="platform:f38"
> PRETTY_NAME="Fedora Linux 38 (Xfce)"
> ANSI_COLOR="0;38;2;60;110;180"
> LOGO=fedora-logo-icon
> CPE_NAME="cpe:/o:fedoraproject:fedora:38"
> DEFAULT_HOSTNAME="fedora"
> HOME_URL="https://fedoraproject.org/"
> DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f38/system-administrators-guide/"
> SUPPORT_URL="https://ask.fedoraproject.org/"
> BUG_REPORT_URL="https://bugzilla.redhat.com/"
> REDHAT_BUGZILLA_PRODUCT="Fedora"
> REDHAT_BUGZILLA_PRODUCT_VERSION=38
> REDHAT_SUPPORT_PRODUCT="Fedora"
> REDHAT_SUPPORT_PRODUCT_VERSION=38
> SUPPORT_END=2024-05-14
> VARIANT="Xfce"
> VARIANT_ID=xfce
> 

[liveuser@localhost-live]$ `uname -a`

> Linux localhost-live 6.3.7-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Jun  9 15:21:11 UTC 2023 x86_64 GNU/Linux

[liveuser@localhost-live]$ `lscpu`

> Architecture:            x86_64
>   CPU op-mode(s):        32-bit, 64-bit
>   Address sizes:         39 bits physical, 48 bits virtual
>   Byte Order:            Little Endian
> CPU(s):                  8
>   On-line CPU(s) list:   0-7
> Vendor ID:               GenuineIntel
>   Model name:            Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
>     CPU family:          6
>     Model:               142
>     Thread(s) per core:  2
>     Core(s) per socket:  4
>     Socket(s):           1
>     Stepping:            10
>     CPU(s) scaling MHz:  61%
>     CPU max MHz:         3400.0000
>     CPU min MHz:         400.0000
>     BogoMIPS:            3600.00
>     Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2
>                           ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology 
>                          nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm
>                           pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowpref
>                          etch cpuid_fault epb invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad 
>                          fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsav
>                          ec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d arch_ca
>                          pabilities
> Virtualization features: 
>   Virtualization:        VT-x
> Caches (sum of all):     
>   L1d:                   128 KiB (4 instances)
>   L1i:                   128 KiB (4 instances)
>   L2:                    1 MiB (4 instances)
>   L3:                    6 MiB (1 instance)
> NUMA:                    
>   NUMA node(s):          1
>   NUMA node0 CPU(s):     0-7
> Vulnerabilities:         
>   Itlb multihit:         KVM: Mitigation: VMX disabled
>   L1tf:                  Mitigation; PTE Inversion; VMX conditional cache flushes, SMT vulnerable
>   Mds:                   Mitigation; Clear CPU buffers; SMT vulnerable
>   Meltdown:              Mitigation; PTI
>   Mmio stale data:       Mitigation; Clear CPU buffers; SMT vulnerable
>   Retbleed:              Mitigation; IBRS
>   Spec store bypass:     Mitigation; Speculative Store Bypass disabled via prctl
>   Spectre v1:            Mitigation; usercopy/swapgs barriers and __user pointer sanitization
>   Spectre v2:            Mitigation; IBRS, IBPB conditional, STIBP conditional, RSB filling, PBRSB-eIBRS Not affected
>   Srbds:                 Mitigation; Microcode
>   Tsx async abort:       Not affected
> 

[root@localhost-live]# `/run/media/root/05_May_2023/aircrack-ng/scripts/airmon-ng.linux` 

> PHY	Interface	Driver		Chipset
> phy0	wlo1		iwlwifi		Intel Corporation Dual Band Wireless-AC 3168NGW [Stone Peak] (rev 10)
> 

## Aircrack-ng version
Aircrack-ng 1.7
- Version: 1.7
- Commit Revision hash: most recent commit: [d4e7705](https://github.com/aircrack-ng/aircrack-ng/commit/d4e7705e759d5bd163805ceae02d72b7d57d29e1)

## Defect
The output of (GNU) `find /sys/bus/pci/devices`, as a complete command, currently will include the search directory, so a count of newlines from within the output of `find` will always return a count of at least one, and the test for equivalence to zero will always return true.  The same error appears to occur on line 137.   

A `grep` of the rest of the directory, 'aircrack-ng/scripts', found the string 'find' only occurs in this script in comments, as arguments to other commands or as a sub-string within function names, and does not occur in either airmon-ng.freebsd or airodump-ng-oui-update. 

Could any of the files within /sys/bus/pci/devices (line 134) or /sys/bus/pci_express/devices (line 137) ever contain newlines? Adding '-type l' might be enough. 

```sh
[root@localhost-live aircrack-ng]# POSIXLY_CORRECT=Y sh
sh-5.2# find /sys/bus/pci/devices -type l
/sys/bus/pci/devices/0000:03:00.0
/sys/bus/pci/devices/0000:00:17.0
/sys/bus/pci/devices/0000:00:1f.2
/sys/bus/pci/devices/0000:00:1c.0
/sys/bus/pci/devices/0000:00:15.1
/sys/bus/pci/devices/0000:00:1f.0
/sys/bus/pci/devices/0000:02:00.0
/sys/bus/pci/devices/0000:00:1d.1
/sys/bus/pci/devices/0000:00:13.0
/sys/bus/pci/devices/0000:00:04.0
/sys/bus/pci/devices/0000:00:16.0
/sys/bus/pci/devices/0000:01:00.0
/sys/bus/pci/devices/0000:00:1f.3
/sys/bus/pci/devices/0000:00:00.0
/sys/bus/pci/devices/0000:00:15.0
/sys/bus/pci/devices/0000:00:1d.0
/sys/bus/pci/devices/0000:00:14.2
/sys/bus/pci/devices/0000:00:02.0
/sys/bus/pci/devices/0000:00:14.0
/sys/bus/pci/devices/0000:00:1f.4
sh-5.2# find /sys/bus/pci/devices 
/sys/bus/pci/devices
/sys/bus/pci/devices/0000:03:00.0
/sys/bus/pci/devices/0000:00:17.0
/sys/bus/pci/devices/0000:00:1f.2
/sys/bus/pci/devices/0000:00:1c.0
/sys/bus/pci/devices/0000:00:15.1
/sys/bus/pci/devices/0000:00:1f.0
/sys/bus/pci/devices/0000:02:00.0
/sys/bus/pci/devices/0000:00:1d.1
/sys/bus/pci/devices/0000:00:13.0
/sys/bus/pci/devices/0000:00:04.0
/sys/bus/pci/devices/0000:00:16.0
/sys/bus/pci/devices/0000:01:00.0
/sys/bus/pci/devices/0000:00:1f.3
/sys/bus/pci/devices/0000:00:00.0
/sys/bus/pci/devices/0000:00:15.0
/sys/bus/pci/devices/0000:00:1d.0
/sys/bus/pci/devices/0000:00:14.2
/sys/bus/pci/devices/0000:00:02.0
/sys/bus/pci/devices/0000:00:14.0
/sys/bus/pci/devices/0000:00:1f.4
sh-5.2# 
exit
[root@localhost-live aircrack-ng]# POSIXLY_CORRECT=Y sh
sh-5.2# find /sys/bus/pci_express/devices
/sys/bus/pci_express/devices
/sys/bus/pci_express/devices/0000:00:1d.0:pcie002
/sys/bus/pci_express/devices/0000:00:1d.0:pcie010
/sys/bus/pci_express/devices/0000:00:1d.1:pcie001
/sys/bus/pci_express/devices/0000:00:1c.0:pcie001
/sys/bus/pci_express/devices/0000:00:1d.0:pcie001
/sys/bus/pci_express/devices/0000:00:1d.1:pcie002
/sys/bus/pci_express/devices/0000:00:1c.0:pcie002
/sys/bus/pci_express/devices/0000:00:1d.1:pcie010
/sys/bus/pci_express/devices/0000:00:1c.0:pcie010
sh-5.2# find /sys/bus/pci_express/devices -type l
/sys/bus/pci_express/devices/0000:00:1d.0:pcie002
/sys/bus/pci_express/devices/0000:00:1d.0:pcie010
/sys/bus/pci_express/devices/0000:00:1d.1:pcie001
/sys/bus/pci_express/devices/0000:00:1c.0:pcie001
/sys/bus/pci_express/devices/0000:00:1d.0:pcie001
/sys/bus/pci_express/devices/0000:00:1d.1:pcie002
/sys/bus/pci_express/devices/0000:00:1c.0:pcie002
/sys/bus/pci_express/devices/0000:00:1d.1:pcie010
/sys/bus/pci_express/devices/0000:00:1c.0:pcie010
sh-5.2# 
```

### Details

```sh
[liveuser@localhost-live]$ find /sys/bus/pci/devices 
/sys/bus/pci/devices
/sys/bus/pci/devices/0000:03:00.0
/sys/bus/pci/devices/0000:00:17.0
# <...>
/sys/bus/pci/devices/0000:00:1f.4
[liveuser@localhost-live]$ find /sys/bus/pci/devices \! -name '0000*'
/sys/bus/pci/devices

[liveuser@localhost-live]$ set -x ; [ "$(find /sys/bus/pci/devices \! -name '0000*' | wc -l)" != '0' ] ; echo $? ; set -
++ find /sys/bus/pci/devices '!' -name '0000*'
++ wc -l
+ '[' 1 '!=' 0 ']'
+ echo 0
0
+ set -
```

**Comment by ZeroChaos-** [comment]
looks good to me, thanks!

**Comment by wileyhy** [comment]
You've got it!

**Comment by wileyhy** [comment]
Resolved by merge of PR #2557
